qiita_pet/test/test_qiita_redbiom.py

```diff
@@ -0,0 +1,161 @@
+from unittest import main
+
+from json import loads
+
+from qiita_pet.test.tornado_test_base import TestHandlerBase
+
+
+class TestRedbiom(TestHandlerBase):
+
+    def test_get(self):
+        response = self.get('/redbiom/')
+        self.assertEqual(response.code, 200)
+
+    def test_post_metadata(self):
+        post_args = {
+            'search': 'Diesel',
+            'search_on': 'metadata'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success', 'message': '', 'data': OBSERVATION}
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': 'inf',
+            'search_on': 'metadata'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success',
+               'message': 'No samples where found! Try again ...', 'data': []}
+        self.assertEqual(loads(response.body), exp)
+
+        post_args = {
+            'search': '4353076',
+            'search_on': 'metadata'
+        }
+        response = self.post('/redbiom/', post_args)
+        self.assertEqual(response.code, 200)
+        exp = {'status': 'success',
+               'message': ('Not a valid search: "4353076", are you sure this '
```

Comment:
not exactly. we intentionally do not index things which look like numbers as i could not determine a reasonable strategy for determining when a number was or was not a good thing to index. in general, it is not and it would have very large impact on the key space. however, as a downside, important numbers like host taxon ID are not index. but, if a user knows they want to search for host taxon ID, they can still do that with `where host_taxid=foo`